File: semestr-2/jipp/zadania/lab6/BinomialSolver.cpp
#include "stdafx.h"
#include "BinomialSolver.h"

double BinomialSolver::getA() const {
	return a;
}

double BinomialSolver::getB() const {
	return b;
}

double BinomialSolver::getC() const {
	return c;
}

double BinomialSolver::getX1() const {
	return x1;
}

double BinomialSolver::getX2() const {
	return x2;
}

double BinomialSolver::calculate(double x) const {
	return a * x * x + b * x + c;
}

File: semestr-2/jipp/java/lab11/src/com/company/BinomialSolver.java
package com.company;

public class BinomialSolver {
    private double a, b, c;
    private double x1, x2;

    public BinomialSolver(double a, double b, double c) {
        this.a = a;
        this.b = b;
        this.c = c;
        double delta;
        delta = Math.pow(b, 2) - 4*a*c;
        if (delta < 0) throw new ArithmeticException();
        else {
            x1 = (-b - Math.sqrt(delta))/(2*a);
            x2 = (-b + Math.sqrt(delta))/(2*a);
        }
    }

    public double getA() {
        return a;
    }

    public double getB() {
        return b;
    }

    public double getC() {
        return c;
    }

    public double getX1() {
        return x1;
    }

    public double getX2() {
        return x2;
    }

    public double calculate(double x) {
        return a*Math.pow(x, 2) + b*x + c;
    }
}
